stream/stream_libarchive.c

```diff
@@ -207,6 +207,12 @@ static char **find_volumes(struct stream *primary_stream)
     struct bstr base = bstr_splice(primary_url, 0, -strlen(pattern->match));
     for (int i = pattern->start; i <= pattern->stop; i++) {
         char* url = pattern->volume_url(res, pattern->format, base, i);
+        FILE *f = fopen(url, "r");
```

Comment:
[access(2)](https://linux.die.net/man/2/access)

Reply:
That's definitely smarter.